### PR TITLE
Add remote source sending feature to IoT.js

### PR DESCRIPTION
**Pull request by DanielBallaSZTE**
This patch makes it possible to send over client source to IoT.js through the debugger clients.
Note:
 - Currently only one source file is handled. In the future, there will be a follow-up patch, allowing the use of multiple source files through require. This requires them to be ordered properly, since importing a module which has not been compiled yet can cause complications.

IoT.js-DCO-1.0-Signed-off-by: Daniel Balla dballa@inf.u-szeged.hu

### CODE REVIEW

File: src/modules/iotjs_module_process.c [hunk 1]
```diff
@@ -290,7 +328,17 @@ iotjs_jval_t InitProcess() {
...
 
-  SetProcessArgv(process);
+  bool wait_source =
+      iotjs_environment_config(iotjs_environment_get())->debugger_wait_source;
+
+  if (!wait_source) {
+    SetProcessArgv(process);
+  }
+
+  iotjs_jval_t* wait_source_val = iotjs_jval_get_boolean(wait_source);
```
Comment (LaszloLango) [inline src/modules/iotjs_module_process.c]: iotjs_jval_t wait_source_val = *iotjs_jval_get_boolean(wait_source);

**Comment by DanielBallaSZTE** [comment]
@LaszloLango Fixed it, thanks for the review!

**Comment by glistening** [comment]
@DanielBallaSZTE Does debugger feature is enabled by default for both release and debug build?

**Comment by DanielBallaSZTE** [comment]
@glistening You can turn it on with `--jerry-debugger` in both release and debug builds. However, by default it is disabled.

**Comment by glistening** [comment]
@DanielBallaSZTE Okay. Debugger specific function and field are not enclosed with appropriate flags. Is it trivial for both size and performance?

**Comment by DanielBallaSZTE** [comment]
@glistening I see your point, I'll add a check whether debugger is turned on, so the user won't be able to use other debugger flags without having the debugger turned on first.

Performance-wise there shouldn't be any changes at all, the only thing that changed is instead of processing the file argument from `argv` it is being processed after the debugger-client sent it over, but only when `debugger-wait-source` flag is turned on.

Considering size, memory consumption peaks for a moment, when the code is being passed to IoT.js itself, for a moment the code is stored twice in the memory, than it turns back to normal. Code size might be somewhat affected, if you wish, I could make some measurements for you.

File: src/modules/iotjs_module_process.c [hunk 2]
```diff
@@ -290,7 +328,17 @@ iotjs_jval_t InitProcess() {
...
-  SetProcessArgv(process);
+  bool wait_source =
+      iotjs_environment_config(iotjs_environment_get())->debugger_wait_source;
+
+  if (!wait_source) {
+    SetProcessArgv(process);
+  }
+
+  iotjs_jval_t wait_source_val = *iotjs_jval_get_boolean(wait_source);
+  iotjs_jval_set_property_jval(&process,
```
Comment (LaszloLango) [inline src/modules/iotjs_module_process.c]: Note: This must be updated if #1228 will land first. #1228 will change the prototype of `iotjs_jval_set_property_jval`.

Comment (DanielBallaSZTE) [reply src/modules/iotjs_module_process.c]: Thanks, I'll definitely update if your patch lands.

**Comment by yichoi** [comment]
@DanielBallaSZTE rebase please

**Comment by glistening** [comment]
@DanielBallaSZTE Oh, I mentioned the size and performance for whole debugger feature, not this PR only. For example, `Config` structure contains several members for debugger, it could be saved if the member definition is enclosed with debugger build macro. If binary increase for debugger is trivial, it is okay, and LGTM.

**Comment by DanielBallaSZTE** [comment]
@glistening Hmm..  In a different patch, if this one lands, I can rethink the structure how debugger features are being handled, and only allocate memory for them, if the debugger itself is turned on.

**Comment by yichoi** [comment]
LGTM

**Pull request merged**